modules/mod_dso.c

```diff
@@ -657,17 +657,21 @@ MODRET set_modulepath(cmd_rec *cmd) {
 
 static void dso_restart_ev(const void *event_data, void *user_data) {
   module *m, *mi;
-#ifdef PR_USE_CTRLS
+#if defined(PR_USE_CTRLS)
   register unsigned int i = 0;
 #endif /* PR_USE_CTRLS */
 
-  if (dso_pool)
+  if (dso_pool != NULL) {
     destroy_pool(dso_pool);
+  }
 
   dso_pool = make_sub_pool(permanent_pool);
   pr_pool_tag(dso_pool, MOD_DSO_VERSION);
 
-#ifdef PR_USE_CTRLS
+  /* Reset the ModulePath (see Issue #1476). */
+  dso_module_path = PR_LIBEXEC_DIR;
```

Comment:
This is the key change/fix here.  Without this, the `dso_module_path` value will be stale, as it is duplicated out of the now-destroyed previous `dso_pool`.